### PR TITLE
New docker bundle

### PR DESCRIPTION
**Pull request by ita-sammann**
I have created a new Docker image and Docker Compose config according to discussion in #184:
 * It is fully configurable by environment variables or by using `.env` file.
 * It downloads the latest (or any other set by `FACTORIO_VERSION` var) version of Factorio on startup, so you don't need to rebuild image to update Factorio.
 * Replaced Nginx with Traefik to handle HTTPS and certificates.
 * Added `docker-compose.simple.yaml` without Traefik and HTTPS if one wants to handle security manually.

### CODE REVIEW

File: docker/.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FACTORIO_VERSION=latest
```
Comment (SuperSandro2000) [inline docker/.env:1]: I would move this into the docker-compose files.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: docker/.env (hunk 1 above)
Comment (ita-sammann) [reply docker/.env]: Done

**Review by knoxfighter — COMMENTED** [review]
Additionally features, that would be great:
- `Update `Dockerfile-build` or `Dockerfile`, that it is possible to generate a docker image from local sourcecode (makes testing and creating docker images for development much easier.)
- remove `build.sh` if not needed for previously mentioned running with local sourcecode.

TODO (not for you, but a message to myself):
- use `mroote/factorio-server-manager` as image.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,29 +1,28 @@
-# glibc is required for Factorio Server binaries to run
-FROM frolvlad/alpine-glibc
+# Glibc is required for Factorio Server binaries to run
+FROM frolvlad/alpine-glibc:alpine-3.12
```
[the remark this answers is not in the document]
Comment (ita-sammann) [reply docker/Dockerfile:2]: No particular reason. Just copy-pasted this version from `factoriotools/factorio-docker` Dockerfile.
Will change to the latest.

[the remark this answers is not in the document]
Comment (ita-sammann) [reply docker/Dockerfile]: Done

File: docker/README.md [hunk 3]
```diff
@@ -1,40 +1,62 @@
 # Factorio Server Manager Docker Image
 
+## Prerequisites
+You need to have [Docker](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-20-04)
+and [Docker Compose](https://docs.docker.com/compose/install/) installed.
+
 ## Getting started?
 
-Pull the Docker container from Docker Hub using the pull command
```
[the remark this answers is not in the document]
Comment (ita-sammann) [reply docker/README.md:5]: Done

**Comment by ita-sammann** [comment]
> Update `Dockerfile-build` or `Dockerfile`, that it is possible to generate a docker image from local sourcecode (makes testing and creating docker images for development much easier.)

I've tried but haven't found an easy way to build a docker image from local source code without moving Dockerfiles to the root of the repository. And I think that moving all this stuff (if it is at all acceptable) should be done in a separate PR.

**Comment by knoxfighter** [comment]
ok, i somehow managed to push changes to your fork :confused: 
These changed will make it possible to build the docker image from local sourcecode. Can you please look into it once, if i haven't missed something. The rest is fine for me know and it works. After you had a look on my last commits i will merge this PR.

File: docker/build.sh [hunk 4]
```diff
@@ -1,8 +1,10 @@
...
-echo "Creating build..."
-make gen_release
-echo "Copying build artifacts..."
-cp -v build/* /build/
\ No newline at end of file
+cd ..
+make build
+cp build/factorio-server-manager-linux.zip docker/factorio-server-manager-linux.zip
+
+cd docker
```
Comment (SuperSandro2000) [inline docker/build.sh:7]:
```suggestion
(
  cd ..
  make build
  cp build/factorio-server-manager-linux.zip docker/factorio-server-manager-linux.zip
)
```

Using a subshell removes the requirement to cd back.

File: docker/Dockerfile-local [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+VOLUME /opt/fsm-data /opt/factorio/saves /opt/factorio/mods /opt/factorio/config
+
+EXPOSE 80/tcp 34197/udp
+
+RUN apk add --no-cache curl tar xz unzip jq
+
+WORKDIR /opt
+
+# Install FSM
+COPY factorio-server-manager-linux.zip /factorio-server-manager-linux.zip
```
[the remark this answers is not in the document]
Comment (knoxfighter) [reply docker/Dockerfile-local:20]: In theory yes, but i dont see the benefit of unzipping it on the host. We just move some commands from one place to another.

**Comment by ita-sammann** [comment]
For some reason, a was fixed on the idea that code must be compiled inside a docker container. Now as I see how it is built outside, it looks very obvious to me :)

Anyway, all changes look good to me.

**Pull request merged**